create.ps1

```diff
@@ -1,194 +1,254 @@
+            }
+            "System.Net.WebException" {
+                $Exception.ErrorMessage = [System.IO.StreamReader]::new(
+                    $ErrorObject.Exception.Response.GetResponseStream()).ReadToEnd()
+                break
+            }
+            default {
+                $Exception.ErrorMessage = $ErrorObject.Exception.Message
+            }
+        }
+
+        $Exception.VerboseErrorMessage = @(
+            "Error at Line [$($ErrorObject.InvocationInfo.ScriptLineNumber)]: $($ErrorObject.InvocationInfo.Line)."
+            "ErrorMessage: $($Exception.ErrorMessage) [$($ErrorObject.ErrorDetails.Message)]"
+        ) -Join ' '
+
+        Write-Output $Exception
     }
 }
-#EndRegion
+#endregion functions
 
-if (-not($dryRun -eq $true)) {
-    try {
-        Write-Verbose 'Getting accessToken'
-        $splatGetTokenParams = @{
-            TenantId     = $config.TenantId
-            ClientId     = $config.ClientId
-            ClientSecret = $config.ClientSecret
-            Scope        = $config.Scope
-        }
-        $accessToken = (Get-LisaAccessToken @splatGetTokenParams).access_token
-        $authorizationHeaders = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
-        $authorizationHeaders.Add("Authorization", "Bearer $accessToken")
-        $authorizationHeaders.Add("Content-Type", "application/json")
-        $authorizationHeaders.Add("Mwp-Api-Version", "1.0")
-
-        # Create user
-        Write-Verbose "Creating KPN Lisa account for '$($personObj.DisplayName)'"
-        $splatParams = @{
-            Uri     = "$($config.BaseUrl)/Users?filter=startswith(userprincipalname,'$($account.userPrincipalName)')"
-            Method  = 'get'
-            Headers = $authorizationHeaders
-        }
-        $userResponse = Invoke-RestMethod @splatParams
-        if ($userResponse.count -eq 0) {
-            $splatParams = @{
-                Uri     = "$($config.BaseUrl)/Users"
-                Method  = 'POST'
-                Body    = ($account | ConvertTo-Json)
-                Headers = $authorizationHeaders
-            }
-            $userResponse = Invoke-RestMethod @splatParams
-            $objectId = $($userResponse.objectId)
-            $auditMessage = "Account '$($personObj.DisplayName)' Created. Id: '$objectId"
-
-            #Set Default WorkSpaceProfile
-            $workSpaceProfileGuid = "500708ea-b69f-4f6c-83fc-dd5f382c308b" #WorkspaceProfile   "friendlyDisplayName": "Ontzorgd"
-
-            $splatParams = @{
-                Uri     = "$($config.BaseUrl)/Users/$objectId/WorkspaceProfiles"
-                Method  = 'PUT'
-                Headers = $authorizationHeaders
-                body    = ($workSpaceProfileGuid | ConvertTo-Json)
+
+#region Aliasses
+$Config = $ActionContext.Configuration
+$Account = $OutputContext.Data
+$AuditLogs = $OutputContext.AuditLogs
+
+$Person = $PersonContext.Person
+$Manager = $PersonContext.Manager
+#endregion Aliasses
+
```

Comment:
- Not going to add `$outputContext.Success`, see other comments
- added `$outputContext.AccountReference` in script
- I don't really see the added value for the debug switch. HelloID removes all the verbose messages when an action is successful, so there is no real need to suppress this in production. Hope to hear your view on this.
- This system is cloudconnector focused, it even requires powershell 7. the cloud powershell is already TLS1.2 enabled so there is also no need for this. (even if the server is used the TLS1.2 should be configured in the register instead of force overwriting it with powershell...)

Reply:
agreed, suggestion:
try {
    # do logic
    # auditlog iserror = $false
}
catch {
    # upon error
    # auditlog iserror = $true
}
finally {
    # Check if auditLogs contains errors, if no errors are found, set success to true
    if (!($outputContext.AuditLogs.IsError -contains $true)) {
        $outputContext.Success = $true
    } 
}

about the debug part, you are correct. We use this for the preview part. But I agree, not critical.
about the TLS part, in this case you are right, but from the readme it is not clear that it only works with the cloud agent.